src/components/Table/head/header.js

```diff
@@ -34,30 +49,57 @@ export default class Header extends Component {
         return null;
     }
 
-    handleMouseUp(event) {
+    handleMouseUp() {
         document.removeEventListener('mouseup', this.handleMouseUp);
         document.removeEventListener('mousemove', this.handleMouseMove);
         const headerContainerWidth = this.headerContainer.current.getBoundingClientRect().width;
-        const width = `${headerContainerWidth + this.newXPosition}px`;
-        const { columnIndex, onResize } = this.props;
-        onResize(event, columnIndex, width);
+        this.resizeBar.current.style.transform = 'unset';
+        const { minColumnWidth, maxColumnWidth } = this.props;
+        const width = headerContainerWidth + this.newXPosition;
+        if (width < minColumnWidth) {
+            this.setColumnWdith(minColumnWidth);
+        } else if (width > maxColumnWidth) {
+            this.setColumnWdith(maxColumnWidth);
+        } else {
+            this.setColumnWdith(width);
+        }
     }
 
     handleMouseMove(event) {
+        const { minColumnWidth, maxColumnWidth } = this.props;
         this.newXPosition = event.clientX - this.startXPosition;
+        const { width } = this.headerContainer.current.getBoundingClientRect();
+        const minXPosition = minColumnWidth - width;
+        const maxXPosition = maxColumnWidth - width;
+        if (this.newXPosition < minXPosition) {
+            this.newXPosition = minXPosition;
+        } else if (this.newXPosition > maxXPosition) {
+            this.newXPosition = maxXPosition;
+        }
         this.resizeBar.current.style.transform = `translateX(${this.newXPosition}px)`;
     }
 
     handleMouseDown(dragEvent) {
         dragEvent.preventDefault();
-        this.startXPosition = dragEvent.clientX;
-        document.addEventListener('mousemove', this.handleMouseMove);
-        document.addEventListener('mouseup', this.handleMouseUp);
+        const { resizeColumnDisabled } = this.props;
+        const isResizable = !resizeColumnDisabled || this.state.width === undefined;
+        if (isResizable) {
+            this.startXPosition = dragEvent.clientX;
```

Comment:
we need to declare:
```this.newXPosition = 0;```
because if I click de resize bar without move it this variable is never declared and then in handleMouseUp when:
```const width = headerContainerWidth + this.newXPosition;```
we got NaN since this.newXPosition is undefined
Also we need to declare it here because whenever we click the resize bar we need to reset this value to 0 since if we move the resize bar 30px to right then this.newXPosition get 30 and when we press the resize bar again without move, it will resize the column with 30 and we don't want this.